server/datastore/mysql/migrations/tables/20231024105026_AddWindowsMDMTables.go

```diff
@@ -0,0 +1,111 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20231024105026, Down_20231024105026)
+}
+
+func Up_20231024105026(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+-- This table contains the commands, which may target multiple devices.
+CREATE TABLE windows_mdm_commands (
+	-- managed and generated by Fleet, and used as CmdID in the MSMDM messages.
+	command_uuid     VARCHAR(127) NOT NULL,
+	-- the raw XML of the command
+	raw_command      MEDIUMTEXT NOT NULL,
+	-- whether the command was generated from the system or from a user
+	system_origin    TINYINT(1) NOT NULL DEFAULT FALSE,
+	-- the target OMADM URI for the command
+	target_loc_uri   VARCHAR(255) NOT NULL,
+
+	created_at       TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+	updated_at       TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+	PRIMARY KEY (command_uuid)
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create windows_mdm_commands table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE windows_mdm_command_queue (
+	host_uuid    VARCHAR(255) NOT NULL,
+	command_uuid VARCHAR(127) NOT NULL,
+
+	-- whether the command should be processed or not (can be deactivated to be ignored)
+	active       BOOLEAN NOT NULL DEFAULT 1,
+
+	created_at   TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+	updated_at   TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+	PRIMARY KEY (host_uuid, command_uuid),
+
+	-- NOTE: no foreign key added for host_uuid, we don't add FKs to the hosts table.
+
+	FOREIGN KEY (command_uuid)
+		REFERENCES windows_mdm_commands (command_uuid)
+		ON DELETE CASCADE ON UPDATE CASCADE
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create windows_mdm_command_queue table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE windows_mdm_responses (
+	id           INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+	-- the host that sent the response
+	host_uuid    VARCHAR(255) NOT NULL,
+	-- the full SyncML, potentially containing results for multiple commands
+	raw_response MEDIUMTEXT NOT NULL,
+
+	PRIMARY KEY (id)
+)`)
+	if err != nil {
+		return fmt.Errorf("failed to create windows_mdm_responses table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+CREATE TABLE windows_mdm_command_results (
+	host_uuid    VARCHAR(255) NOT NULL,
+	command_uuid VARCHAR(127) NOT NULL,
+
+	-- TODO: what is that? If it's the <Status><Data>200</Data></Status> then
+	-- seems like it would be a duplicate of the rx_error_code (now status_code)?
+	status       VARCHAR(31)  NOT NULL,
```

Comment:
My guess is that this field has been superseded by `status_code` below?

Reply:
I think this should instead be

`result   MEDIUMTEXT NOT NULL,`

Let's say that a given `Get` command has the the following `Results` response

```
     <Results>
       <CmdID>3</CmdID>
       <MsgRef>1</MsgRef>
       <CmdRef>5</CmdRef>
       <Item>
         <Source>
           <LocURI>./DevDetail/Ext/Microsoft/DeviceName</LocURI>
         </Source>
         <Data>DESKTOP-AC82RC0</Data>
       </Item>
     </Results>
```

My idea is that the result field will just have the result data, in this case  `DESKTOP-AC82RC0`

Then the `raw_result` field should have the entire Results response (<Result> ... </Result> ) 

Having just the `result` data is helpful to don't avoid parsing the result content if this data is needed.

On the other hand, the `LocURI` of the result should be stored here, too. This will facilitate retrieving the value of a given setting.  Let's say that we need to know the desktop name for a given `DeviceID`, logic can use this table to retrieve the `Results` value for `LocURI` `./DevDetail/Ext/Microsoft/DeviceName` and get this value.

Reply:
Yeah I can see that being useful, however do we have a need for that at the moment? If not I'd suggest we only add this once we definitely need it. I would assume that when we receive results, there would be some server-side processing of those and if we want to ingest some specific data in response of an internal (fleet-initiated) command, we'd grab it when we process the result and store it in the standard Fleet tables, instead of querying it later on in the `windows_mdm_...` tables.

Reply:
yes, we can add it in the future if needed. That's ok to me. I see this table acting as a cache for that information.